### ISSUE TITLE
Custom fields on new file/post creation dialog

### ISSUE THREAD

**Issue by ygd** [zhuochun/md-writer#219]
I think custom fields, which you can configure in `_mdwriter.cson`, would be cool to add more front matter data.

eg. if there would be an additional field called `author` in the new post dialog you could directly fill the author front matter and use it also for composing the title or file name of the post like:

```
// _mdwriter.cson

// add custom field
customNewPostFields:
  author: "Default value"

// use custom field in file name
newPostFileName: "{year}-{month}-{day}-{slug}-{author}{extension}"

// use custom field in front matter
frontMatter: """
---
title: "Quote by {author} about SOME_TAG"
description: "Quote: {title} - {author}"
quote: "{title}"
quote_author: "{author}"
---
"""
```

**Comment by zhuochun** [comment]
Depending on how often you change the author, current supported is add:

```
templateVariables:
  author: "value"
```

you should be able to reference `{author}`, but it is static value as specified.

**Comment by ygd** [comment]
@zhuochun yep, I saw it last time. but having something dynamic would really cool. would be also good base for further features like supporting collections and stuff …

I will have a look into the code, and when I feel I can implement it myself I would open a pull request.

**Comment by theramiyer** [comment]
This would be an awesome feature to have; we'd love to be able to add the fields we want in the New Post dialog! Wish I could help. Unfortunately I'm not a developer. :)

**Comment by zhuochun** [comment]
Released in v2.8.0, wiki: https://github.com/zhuochun/md-writer/wiki/Settings-for-Front-Matters#setup-custom-fields
